pkg/apis/infinispan/v2alpha1/backup_types.go

```diff
@@ -0,0 +1,98 @@
+package v2alpha1
+
+import (
+	v1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BackupSpec defines the desired state of Backup
+type BackupSpec struct {
+	Cluster   string                     `json:"cluster"`
+	Volume    BackupVolumeSpec           `json:"volume,optional,omitempty"`
+	Resources BackupResources            `json:"resources,optional,omitempty"`
+	Container v1.InfinispanContainerSpec `json:"container,optional,omitempty"`
+}
+
+type BackupVolumeSpec struct {
+	Storage          *string `json:"storage,optional,omitempty"`
+	StorageClassName *string `json:"storageClassName,optional,omitempty"`
+}
+
+type BackupResources struct {
+	Caches       []string `json:"caches,optional,omitempty"`
+	CacheConfigs []string `json:"cacheConfigs,optional,omitempty"`
+	Counters     []string `json:"counters,optional,omitempty"`
+	ProtoSchemas []string `json:"protoSchemas,optional,omitempty"`
+	Scripts      []string `json:"scripts,optional,omitempty"`
+}
+
+type BackupPhase string
+
+const (
+	// BackupInitializing means the request has been accepted by the system, but the underlying resources are still
+	// being initialized.
+	BackupInitializing BackupPhase = "Initializing"
+	// BackupInitialized means that all required resources have been initialized
+	BackupInitialized BackupPhase = "Initialized"
+	// BackupRunning means that the backup pod has been created and the backup process initiated on the infinispan server.
+	BackupRunning BackupPhase = "Running"
+	// BackupSucceeded means that the backup process on the server has completed and the backup pod has been terminated.
+	BackupSucceeded BackupPhase = "Succeeded"
+	// BackupFailed means that the backup failed on the infinispan server and the backup pod has terminated.
+	BackupFailed BackupPhase = "Failed"
```

Comment:
In case of `Failed` it should be nice to print a reason of this
```
...
  spec:
    cluster: example-infinispan
  status:
    phase: Failed
    pvc: pvc/example-infinispan-backup
```

Reply:
Any proposals on how best to achieve this in the status?

`status.phaseOutput`?

Reply:
I thought that `Reason` could be field for free message text. Isn't it?

Reply:
Of course ... I forgot that was even there :smile: